### PR TITLE
Upgrade upb to the latest

### LINKED ISSUE #29193: Memory leak in upb in master
https://github.com/protocolbuffers/upb/commit/aa779cd779b660e3e4f6560519827bfb15a7b488 fixes a memory leak in upb.  Master (as of today, as well as 1.45.0) depends on an older version without this fix.

We are using Bazel to build gRPC and caught this using our asan build.  We were successfully able to update upb locally to get it to work again, but would appreciate not carrying that patch long term.

### PR DESCRIPTION
**Pull request by veblush**
To https://github.com/protocolbuffers/upb/commit/bef53686ec702607971bd3ea4d4fefd80c6cc6e8
Fixes #29193

### CODE REVIEW

**Comment by veblush** [comment]
[grpc_build_artifacts_multiplatform](https://fusion2.corp.google.com/invocations/f37ecb36-21a5-4ca3-ac73-c3b635de4b2f): running

File: bazel/grpc_deps.bzl [hunk 1]
```diff
@@ -344,11 +344,11 @@ def grpc_deps():
...
         http_archive(
             name = "upb",
-            sha256 = "1cd33bf607ebc83acf71b6078c1d4361ffa49d647a2ce792a557ae98f75500ad",
-            strip_prefix = "upb-a02d92e0257a35f11d4a58b6a932506cbdbb2f29",
+            sha256 = "d0fe259d650bf9547e75896a1307bfc7034195e4ae89f5139814d295991ba681",
+            strip_prefix = "upb-bef53686ec702607971bd3ea4d4fefd80c6cc6e8",
             urls = [
-                "https://storage.googleapis.com/grpc-bazel-mirror/github.com/protocolbuffers/upb/archive/a02d92e0257a35f11d4a58b6a932506cbdbb2f29.tar.gz",
-                "https://github.com/protocolbuffers/upb/archive/a02d92e0257a35f11d4a58b6a932506cbdbb2f29.tar.gz",
+                "https://storage.googleapis.com/grpc-bazel-mirror/github.com/protocolbuffers/upb/archive/bef53686ec702607971bd3ea4d4fefd80c6cc6e8.tar.gz",
```
Comment (jtattermusch) [inline bazel/grpc_deps.bzl:350]: please run the update_mirror.sh script  to update the mirror.

File: third_party/README.md [hunk 2]
```diff
@@ -118,7 +118,10 @@ Apart from the above steps, please perform the following two steps to generate t
 Since upb is vendored in the gRPC repo, you cannot use submodule to update it. Please follow the steps below.
 
 1. Update third_party/upb directory by running
-   `git subtree pull --squash --prefix=third_party/upb https://github.com/protocolbuffers/upb.git main`
+   - `wget https://github.com/protocolbuffers/upb/archive/refs/heads/main.zip`
+   - `rm -rf  ~/git/grpc/third_party/upb`
```
Comment (jtattermusch) [inline third_party/README.md:122]: note that the `~/git/grpc` part is specific to your personal setup, might be different for others. E.g. set variable GRPC_ROOT= at the beginning so others can easily customize?

**Review by jtattermusch — APPROVED** [review]
LGTM after addressing comments.

File: bazel/grpc_deps.bzl (hunk 1 above)
Comment (veblush) [reply bazel/grpc_deps.bzl:350]: yeah I did it.

**Pull request merged**